src/servers/main.cc

```diff
@@ -69,14 +69,19 @@ namespace {
 int32_t repository_poll_secs_ = 15;
 
 // The HTTP, GRPC and metrics service/s and ports. Initialized to
-// default values and modifyied based on command-line args. Set to -1
```

Comment:
What is the motivation to remove the -1 logic for disabling the protocol?

Reply:
It's confusing to have two ways to control the endpoints and we don't really publicly document that setting port to -1 means disabling the endpoint, so this is for internal use and only using `allow_xxx` parameter will be more clear. Thus I remove it

Reply:
But if the user builds the server with both endpoints, it will always launch both/all endpoints. I don't think removing this because it isn't documented is the best idea. I think it's a nice feature that should instead of documented

Reply:
An example to disable endpoint: https://github.com/triton-inference-server/server/pull/2820/files#diff-e15379ef713d35b70b5e1def42a8b3078083b9c68e2e25cf5ef4957fb52e29aaR86

Won't this be a nicer way to do so? `allow-xxx` is listed as a option so we need to ensure backward compatibility, but `port == -1` is not. Again, the functionality is duplicated and I think it is better to keep either one but not both, since backward compatibility is not a concern.

Reply:
Oh yes, I see what you mean now. Go ahead